flink-yarn-tests/src/test/java/org/apache/flink/yarn/YarnTestBase.java

```diff
@@ -608,10 +611,13 @@ public boolean accept(File dir, String name) {
     public static boolean verifyStringsInNamedLogFiles(
             final String[] mustHave, final ApplicationId applicationId, final String fileName) {
         final List<String> mustHaveList = Arrays.asList(mustHave);
-        final File cwd = new File("target", YARN_CONFIGURATION.get(TEST_CLUSTER_NAME_KEY));
+        final File cwd =
```

Comment:
Why was this failing on 2.10, when it worked on 2.8 and 3.3?

Reply:
@gaborgsomogyi ?

Reply:
No idea, I'm just wondering how it worked until now. This area is just super confusing in general because sometimes `target` but sometimes `target/test/data/` is set in `YarnTestBase.java` w/o my change. At top of this locally different execution path runs than in the CI which makes the problem even harder to explain/debug.

What I can imagine but can't point it out:
* one `pom.xml` change made the test inherit different current directory
* YARN is getting different current directory for some magical reason

Reply:
I think if we want to have control here then we must just forget relative paths. Blindly all the logic which is depending on `TEST_CLUSTER_NAME_KEY` must be just kicked out with hardcoded TMP dir providing full path.